src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -60,5 +65,84 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         /// Returns true is the db connection is to a SQL db
         /// </summary>
         public bool IsAzure { get; set; }
+
+        /// <summary>
+        /// Try to get the DbConnection associated with the given connection type string. 
+        /// </summary>
+        public bool TryGetConnection(string connectionType, out DbConnection connection)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+
+            return ConnectionTypeToConnectionMap.TryGetValue(connectionType, out connection);
+        }
+
+        /// <summary>
+        /// Get a List of all DbConnection instances held by this ConnectionInfo
+        /// </summary>
+        public List<DbConnection> GetAllConnections()
+        {
+            return ConnectionTypeToConnectionMap.Values.ToList();
```

Comment:
We don't need/want the list this generates to be editable, since any changes wouldn't be reflected in the dictionary. My personal preference would be to change these Get* methods to getter-only properties that return IEnumerable<whatever> (or ICollection<>) and just return Map.Values/Map.Keys without newing up a list.